include/caffe/util/math_functions.hpp

```diff
@@ -12,6 +12,18 @@
 
 namespace caffe {
 
+template<typename Dtype>
+Dtype caffe_ipow(Dtype base, int exp) {
```

Comment:
I do understand that in places where pow is called often, this matters. In this solver pow is not call that often though (at most 3 times per iteration and layer with parameters), so optimizing a call that takes 0.1 microseconds with one that takes 0.04 microseconds is complete overkill.